### PR TITLE
feat: add postproc and threshold as PostProcessUDF

### PR DESCRIPTION
**Pull request by s0nicboOm**
Add PostProc

### CODE REVIEW

File: numalogic/udfs/postprocess.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        thresh_artifact = _load_model(
+            skeys=keys, dkeys=[thresh_cfg.name], payload=payload, model_registry=self.model_registry
+        )
+        postproc_clf = self.postproc_factory.get_instance(postprocess_cfg)
+
+        #  Postprocess payload
+        if payload.status in (Status.ARTIFACT_FOUND, Status.ARTIFACT_STALE):
+            if thresh_artifact:
+                thresh_clf = thresh_artifact.artifact
+                try:
```
Comment (ab93) [inline numalogic/udfs/postprocess.py:77]: Can we avoid so many nested ifs?

File: numalogic/udfs/postprocess.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        kwargs: Additional arguments
+
+        Returns
+        -------
+        Output data
+        """
+        _start_time = time.perf_counter()
+        try:
+            y_score = model.score_samples(input_)
+        except RuntimeError as err:
```
Comment (ab93) [inline numalogic/udfs/postprocess.py:144]:
```suggestion
        except Exception as err:
```

File: numalogic/udfs/postprocess.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            )
+            messages.append(Message(keys=keys, value=train_payload.to_json(), tags=["train"]))
+        _LOGGER.debug(
+            "%s -  Time taken in postprocess: %.4f sec",
+            payload.uuid,
+            time.perf_counter() - _start_time,
+        )
+        return messages
+
+    def compute(self, model: artifact_t, input_: NDArray[float], **kwargs) -> NDArray[float]:
```
Comment (ab93) [inline numalogic/udfs/postprocess.py:127]:
```suggestion
    def compute(self, model: artifact_t, input_: NDArray[float], postproc_clf=None, **_) -> NDArray[float]:
```

File: numalogic/udfs/postprocess.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        Output data
+        """
+        _start_time = time.perf_counter()
+        try:
+            y_score = model.score_samples(input_)
+        except RuntimeError as err:
+            raise RuntimeError("Threshold model scoring failed") from err
+        try:
+            win_score = np.mean(y_score, axis=0)
+            score = kwargs["postproc_clf"].transform(win_score)
```
Comment (ab93) [inline numalogic/udfs/postprocess.py:148]:
```suggestion
            score = postproc_clf.transform(win_score)
```

File: tests/udfs/test_postprocess.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+from numalogic.udfs.preprocess import PreprocessUDF
+from tests.udfs.utility import input_json_from_file, store_in_redis
+
+logging.basicConfig(level=logging.DEBUG)
+REDIS_CLIENT = FakeStrictRedis(server=FakeServer())
+KEYS = ["service-mesh", "1", "2"]
+DATUM = input_json_from_file(os.path.join(TESTS_DIR, "udfs", "resources", "data", "stream.json"))
+
+
+class TestPostProcessUDF(unittest.TestCase):
```
Comment (ab93) [inline tests/udfs/test_postprocess.py:31]: Can you make the tests independent of all other udfs? Look at the inference tests

Comment (s0nicboOm) [reply tests/udfs/test_postprocess.py:31]: Any reason for it?
The reason I did it this way was to simulate all edge cases like thresh model absent, infer model stale etc etc cause mocking these may not be possible.

File: numalogic/udfs/postprocess.py (hunk 1 above)
Comment (s0nicboOm) [reply numalogic/udfs/postprocess.py:77]: Sure

File: tests/udfs/test_postprocess.py (hunk 5 above)
Comment (ab93) [reply tests/udfs/test_postprocess.py:31]: Because unit tests should be as independent as possible. You can test for all possible edge cases by just changing the input payload.
For the whole pipeline we need to add integration tests

Comment (s0nicboOm) [reply tests/udfs/test_postprocess.py:31]: Gotcha! Will fix it

File: numalogic/udfs/postprocess.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+                    header=Header.MODEL_INFERENCE,
+                )
+                _LOGGER.info(
+                    "%s - Successfully post-processed, Keys: %s, Metrics: %s, x_scaled: %s",
+                    payload.uuid,
+                    keys,
+                    payload.metrics,
+                    list(processed_data),
+                )
+                messages.append(Message(keys=keys, value=payload.to_json(), tags=["output"]))
```
Comment (ab93) [inline numalogic/udfs/postprocess.py:98]: Can you move this to else?

Comment (s0nicboOm) [reply numalogic/udfs/postprocess.py:107]: good catch!

**Review by ab93 — APPROVED** [review]
